src/project/project-actions.tsx

```diff
@@ -454,6 +474,10 @@ export class ProjectActions {
       type: "download-main-file",
     });
 
+    if (this.isDefaultProjectName()) {
+      await this.editProjectName(true);
+    }
```

Comment:
Should this have the same cancel logic as above?

Maybe just have a really explicit method like "ensureProjectNameOrCancel" called for both?

Reply:
[`2cf98b7`](https://github.com/microbit-foundation/python-editor-next/pull/763/commits/2cf98b77f736e0600c02f978367491783250b2f4).